File: ui_rag_next/lib/contracts.ts
export interface ServiceStatus {
  ok: boolean;
  status: number;
}

export interface QueryRequest {
  query: string;
  conversation_id?: string;
}

export type QueryState =
  | "streaming"
  | "retrieving"
  | "answered"
  | "insufficient_context"
  | "unsupported";

export type QueryStreamEventName =
  | "start"
  | "retrieval"
  | "content"
  | "citation"
  | "done";

export interface Citation {
  document_id: string;
  chunk_id: string;
  snippet?: string;
}

export interface APIError {
  code: string;
  message: string;
}

export interface QueryResponse {
  state: QueryState;
  event?: QueryStreamEventName;
  kind?: string;
  answer?: string;
  citations?: Citation[];
  conversation_id?: string;
  error?: APIError;
}

export interface QueryStreamMessage {
  event?: QueryStreamEventName;
  id?: string;
  frame: QueryResponse;
}

export interface IngestDocumentRequest {
  path: string;
}

export type DocumentStatus = "pending" | "indexing" | "ready" | "error" | "outdated";

export interface Document {
  id: string;
  collection_id: string;
  filename: string;
  path: string;
  status: DocumentStatus;
  chunks_count: number;
  error_message?: string;
  created_at: string;
  updated_at: string;
}

export type IngestDocumentState = "indexed" | "unindexed" | "unsupported";

export interface IngestDocumentResponse {
  state: IngestDocumentState;
  document_id?: string;
  citations?: Citation[];
  error?: APIError;
}

export interface ResourceRecord {
  id: string;
  [key: string]: unknown;
}

export type Agent = ResourceRecord;
export type Collection = ResourceRecord;
export type Conversation = ResourceRecord;

export type ResourceName =
  | "agents"
  | "collections"
  | "documents"
  | "conversations";

File: service/internal/domain/contracts.go
package domain

const (
	QueryStateAnswered            = "answered"
	QueryStateInsufficientContext = "insufficient_context"
	QueryStateRetrieving          = "retrieving"
	QueryStateStreaming           = "streaming"
	QueryStateUnsupported         = "unsupported"
	QueryStreamEventStart         = "start"
	QueryStreamEventRetrieval     = "retrieval"
	QueryStreamEventContent       = "content"
	QueryStreamEventCitation      = "citation"
	QueryStreamEventDone          = "done"
	QueryStreamKindLifecycle      = "lifecycle"
	QueryStreamKindRetrieval      = "retrieval"
	QueryStreamKindContent        = "content"
	QueryStreamKindCitation       = "citation"
	QueryStreamKindCompletion     = "completion"
	DocumentStateIndexed          = "indexed"
	DocumentStateUnindexed        = "unindexed"
	DocumentStateUnsupported      = "unsupported"
	ReadinessCodeUnavailable      = "unavailable"
)

type HealthResponse struct {
	Alive   bool   `json:"alive"`
	Service string `json:"service"`
}

type DependencyReason struct {
	Dependency string `json:"dependency"`
	Code       string `json:"code"`
	Detail     string `json:"detail,omitempty"`
}

type ReadinessResponse struct {
	Ready   bool               `json:"ready"`
	Reasons []DependencyReason `json:"reasons,omitempty"`
}

type QueryRequest struct {
	Query          string `json:"query"`
	ConversationID string `json:"conversation_id,omitempty"`
}

type IngestRequest struct {
	Path string `json:"path"`
}

type Citation struct {
	DocumentID string `json:"document_id"`
	ChunkID    string `json:"chunk_id"`
	Snippet    string `json:"snippet,omitempty"`
}

type APIError struct {
	Code    string `json:"code"`
	Message string `json:"message"`
}

type QueryResponse struct {
	State          string     `json:"state"`
	Event          string     `json:"event,omitempty"`
	Kind           string     `json:"kind,omitempty"`
	Answer         string     `json:"answer,omitempty"`
	Citations      []Citation `json:"citations,omitempty"`
	ConversationID string     `json:"conversation_id,omitempty"`
	Error          *APIError  `json:"error,omitempty"`
}

type IngestResponse struct {
	State      string     `json:"state"`
	DocumentID string     `json:"document_id,omitempty"`
	Citations  []Citation `json:"citations,omitempty"`
	Error      *APIError  `json:"error,omitempty"`
}
